crates/rpc/src/server/accept.rs

```diff
@@ -0,0 +1,188 @@
+use std::task::{Context as StdContext, Poll};
+
+use futures::{FutureExt, future::BoxFuture};
+use http::header::ACCEPT;
+use nom::{
+    IResult,
+    bytes::complete::{tag, take_until},
+};
+use semver::{Version, VersionReq};
+use tower::{Layer, Service};
+use tracing::debug;
+
+use crate::COMPONENT;
+
+/// Layer responsible for handling HTTP ACCEPT headers.
+#[derive(Clone)]
+pub struct AcceptLayer {
+    version: VersionReq,
+}
+
+impl AcceptLayer {
+    /// Create a new accept layer with the specified version requirement.
+    ///
+    /// # Panics:
+    ///
+    /// Panics if the version string in Cargo.toml is not valid semver.
+    /// The version string is made into an env var at compile time which means
+    /// that the unit tests prove this cannot panic in practice.
+    pub fn new() -> Self {
```

Comment:
nit: we usually wrap comments after 100 chars.

Reply:
Sorry I don't follow what you mean here. The lines are all under 100 characters long